corgi/tasks/prod_defs.py

```diff
@@ -186,10 +186,19 @@ def parse_variants_from_brew_tags(
         )
         for brew_tag in brew_tags.keys():
             # Also match brew tags in prod_defs with those from ET
-            trimmed_brew_tag = brew_tag.removesuffix("-released")
+            tag_no_candidate = brew_tag.removesuffix("-candidate")
+            tag_no_released = brew_tag.removesuffix("-released")
```

Comment:
Can we just have one trimmed_brew_tag that we need to check for each original brew tag that we process? Like below:
```suggestion
            trimmed_brew_tag = brew_tag.removesuffix("-container-released")
            trimmed_brew_tag = trimmed_brew_tag.removesuffix("-candidate")
            trimmed_brew_tag = trimmed_brew_tag.removesuffix("-released")
            # And so on, for any other suffixes we want to remove
```
I'm assuming here that either a tag doesn't have multiple suffixes, like "-candidate-released" or "-released-candidate". Or if a tag does have multiple suffixes, we probably want to strip all of them, right?

Reply:
The special case is `rhaos-4.10-rhel-8` which has the brew tag 'rhaos-4.10-rhel-8-candidate' in ET product version `{ET_URL}/products/79/product_versions/1625`, but we want to match it against builds with the tag 'rhaos-4.10-rhel-8-container-released' (without the whole `container-released` suffix.)

If we always strip `container-released` first however that won't match the `rhacm-2.4-rhel7` case which we need to strip the `-released` suffix only to match what we have in Collector models `rhacm-2.4-rhel7-container`.

Reply:
Ack, thanks for giving that example! This makes sense and LGTM. It's a shame our data is this complicated but oh well :shrug: